### PR TITLE
[#IOCOM-1062] Add new function CreateRCConfiguration

### PR DESCRIPTION
**Pull request by Garma00**
#### List of Changes

Add a new function to create new remote content configuration with a POST

#### Motivation and Context

We need to expose an api to create new remote content configuration

#### How Has This Been Tested?

Unit and integration tests

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by michaeldisaro — CHANGES_REQUESTED** [review]

File: CreateRCConfiguration/__tests__/handler.test.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      )(aNewRemoteContentConfiguration)
+    );
+    expect(E.isRight(r)).toBeTruthy();
+    if (E.isRight(r))
+      expect(E.isRight(Ulid.decode(r.right.configurationId))).toBeTruthy();
+  });
+});
+
+describe("createRCConfigurationHandler", () => {
+  test("should return 500 if the model return an error", async () => {
```
[the remark this answers is not in the document]
Comment (Garma00) [reply CreateRCConfiguration/__tests__/handler.test.ts:34]: This function does not handle any bad request.
The bad request is handled by the middleware `RequiredBodyPayloadMiddleware` which is defined in `io-functions-commons` and it has it's own unit test.
Actually the bad request path for this function is tested via integration test.

File: openapi/index.yaml [hunk 2]
```diff
@@ -132,9 +307,16 @@ definitions:
         type: string
         format: uri
         description: >-
-          An absolute URI that identifies the specific occurrence of the problem.
+          An absolute URI that identifies the specific occurrence of the
+          problem.
```
[the remark this answers is not in the document]
Comment (Garma00) [reply openapi/index.yaml:311]: Ditto cc @acuiuli

File: openapi/index.yaml [hunk 3]
```diff
@@ -63,11 +63,189 @@ paths:
...
+      hasPrecondition:
+        type: string
+        x-extensible-enum:
+          - ALWAYS
+          - ONCE
+          - NEVER
+      disableLollipopFor:
+        type: array
+        items:
+          type: string
```
[the remark this answers is not in the document]
Comment (Garma00) [reply openapi/index.yaml:129]: This is related to `io-functions-commons` cc @acuiuli

File: openapi/index.yaml [hunk 4]
```diff
@@ -63,11 +63,189 @@ paths:
...
+        - in: body
+          name: body
+          required: true
+          schema:
+            $ref: '#/definitions/NewRCConfiguration'
+      responses:
+        '201':
+          description: Configuration created
+          schema:
+            $ref: '#/definitions/RCConfigurationId'
```
[the remark this answers is not in the document]
Comment (Garma00) [reply openapi/index.yaml:85]: I preferred to take in example the CreateMessage api where we return 201 with the generated id in response for consistency, but we can also return the whole configuration

File: CreateRCConfiguration/handler.ts [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+import { NonEmptyString, Ulid } from "@pagopa/ts-commons/lib/strings";
+import { ObjectIdGenerator } from "@pagopa/io-functions-commons/dist/src/utils/strings";
+import { NewRCConfiguration } from "../generated/definitions/NewRCConfiguration";
+import { RequiredUserIdMiddleware } from "../utils/middlewares";
+
+export const getNewRCConfigurationWithConfigurationId = (
+  generateConfigurationId: ObjectIdGenerator,
+  userId: NonEmptyString
+) => (newRCConfiguration: NewRCConfiguration): RCConfiguration => ({
+  configurationId: (generateConfigurationId() as unknown) as Ulid,
```
[the remark this answers is not in the document]
Comment (Garma00) [reply CreateRCConfiguration/handler.ts:30]: Take a look here @michaeldisaro [Add new ulid generator that actually return a Ulid](https://github.com/pagopa/io-functions-commons/pull/364/commits/8253a6f51afe0e8c0c4a9fb81370758f3bc72545)

File: openapi/index.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (Garma00) [reply openapi/index.yaml:85]: Now the function return the whole configuration in response 😄

File: CreateRCConfiguration/handler.ts [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+        generateConfigurationId,
+        userId,
+        newRCConfiguration
+      )
+    ),
+    TE.map(
+      flow(retrievedRCConfigurationToPublic, publicConfiguration =>
+        ResponseSuccessRedirectToResource(
+          publicConfiguration,
+          `/api/v1/remote-contents/configurations/`,
```
Comment (michaeldisaro) [inline CreateRCConfiguration/handler.ts:142]: Why redirecting?

Comment (Garma00) [reply CreateRCConfiguration/handler.ts:142]: Actually it is not a redirect, this function just set the header `Location`.
The same thing was made with the `CreateMessage` so I used it here for consistency

**Review by michaeldisaro — APPROVED** [review]

**Pull request merged**